sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -140,7 +140,8 @@ abstract class Optimizer(sessionCatalog: SessionCatalog, conf: CatalystConf)
   object OptimizeSubqueries extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan transformAllExpressions {
       case s: SubqueryExpression =>
-        s.withNewPlan(Optimizer.this.execute(s.plan))
+        val ReturnAnswer(newPlan) = Optimizer.this.execute(ReturnAnswer(s.plan))
```

Comment:
I can rename it if you think that is better. It seems a bit silly to me to have two logical plan nodes for this.